src/subclass/mod.rs

```diff
@@ -189,9 +189,23 @@
 //!     }
 //! }
 //!
-//! pub fn main() {
+//! // Optionally, define a wrapper type to make it more ergononic to use from Rust
+//! glib_wrapper! {
+//!     pub struct SimpleObject(ObjectSubclass<SimpleObjectPrivate, SimpleObjectClass>);
```

Comment:
I think this is indeed an improvement. The main problem that is left is that users will have to list all the parent classes and implemented interfaces nonetheless, but I guess that's just how it is :)

Instead of `ObjectSubclass` there might be a better name, not sure. The same thing would also work for new interfaces defined in Rust. I can't think of anything better though :)

Reply:
The implementation assumes the type implements `ObjectSubclass`, so it seems like an accurate enough name. But if anyone things of something better, that's fine.

I don't really like the `glib_wrapper!` macro. Of course complicated `macro_rules!` macros end up with macro hygiene issues, an incoherent implementation, and confusing error messages. But I also don't like the fact the macro is used by defining a tuple struct containing a certain 'type', which isn't actually a type and isn't actually the representation of the resulting struct...

But coming up with a better API isn't exactly simple either...

Reply:
> The implementation assumes the type implements ObjectSubclass, so it seems like an accurate enough name. But if anyone things of something better, that's fine.

That's a good point, let's keep it like that then. For interfaces we can add another variant later :)

> I don't really like the glib_wrapper! macro.

Me neither but I couldn't come up with anything better yet... so if you have any ideas go ahead